packages/discovery-provider/src/tasks/entity_manager/entity_manager.py

```diff
@@ -643,6 +663,8 @@ def collect_entities_to_fetch(update_task, entity_manager_txs):
                     )
             if entity_type == EntityType.PLAYLIST:
                 entities_to_fetch[EntityType.PLAYLIST_ROUTE].add(entity_id)
+            if entity_type == EntityType.EVENT:
+                entities_to_fetch[EntityType.EVENT].add(entity_id)
```

Comment:
You have logic in the validation functions to check if related entities such as playlist/track exist. Those have to be added here in order for them to exist when we get to validation.